### DISCUSSION TITLE
Reverse Columns

### DISCUSSION THREAD

**Discussion by connormcmk** [nushell/nushell#6547] [Q&A]
Let's say I have this table:
```
"a:b:c" | split columns ":"
```

How can I reverse the order of the columns? Expected output: 
|column1|column2|column3|
|-|-|-|
|c|b|a|

Is there a way to make it work even if the number of elements changes? E.g. to `a:b:c:d:e`? 

|column1|column2|column3|column4|column5|
|-|-|-|-|-|
|a|b|c|d|e|

Note: please assume that the table is already built. I.e. `"a:b:c:d:e" | reverse | split column ":"` is not an answer

**Comment by fdncred** [comment]
Maybe something like this? 
```
"a:b:c:d:e" | split row ":" | reverse | str join ":" | split column ":"
╭───┬─────────┬─────────┬─────────┬─────────┬─────────╮
│ # │ column1 │ column2 │ column3 │ column4 │ column5 │
├───┼─────────┼─────────┼─────────┼─────────┼─────────┤
│ 0 │ e       │ d       │ c       │ b       │ a       │
├───┼─────────┼─────────┼─────────┼─────────┼─────────┤
│ # │ column1 │ column2 │ column3 │ column4 │ column5 │
╰───┴─────────┴─────────┴─────────┴─────────┴─────────╯
```

**Reply by connormcmk** [reply]
That conversion to and from strings has to be inefficient, right?

**Reply by fdncred** [reply]
I don't think it's converting to and from strings. It's splitting a string and then joining a string and splitting again. Maybe that's what you meant though. :) That was the only way I can think of doing it. It seems like it could be slow considering all the splitting and joining going on. I haven't measured it though. It would be nice to have a command that does that, although I'm not sure I recall anyone asking for this specific thing.

**Comment by incansvl** [comment]
How about-

`"a:b:c" | split column ":" | transpose -i | sort -r | transpose -i`

I think that works for an arbitrary number of columns. E.g:

```
"a:b:c:d:e:f:g:h:i:j:k" | split column ":" | transpose -i | sort -r | transpose -i

│ # │ column0 │ column1 │ column2 │ column3 │ column4 │ column5 │ column6 │ column7 │ column8 │ column9 │ column10 │
├───┼─────────┼─────────┼─────────┼─────────┼─────────┼─────────┼─────────┼─────────┼─────────┼─────────┼──────────┤
│ 0 │ k       │ j       │ i       │ h       │ g       │ f       │ e       │ d       │ c       │ b       │ a        │
╰───┴─────────┴─────────┴─────────┴─────────┴─────────┴─────────┴─────────┴─────────┴─────────┴─────────┴──────────╯
```
